orttraining/orttraining/training_ops/cuda/optimizer/lamb.cu

```diff
@@ -519,11 +577,20 @@ void LambMultiTensorReductionFunctor<TIn1, TIn2, TOut1, TOut2, TBuf>::operator()
   ORT_ENFORCE(thread_count % GPU_WARP_SIZE == 0);
   ORT_ENFORCE((thread_count & (thread_count - 1)) == 0);
 
-  LambMultiTensorReductionImpl<TIn1, TIn2, TOut1, TOut2, TBuf><<<chunk_group.chunk_count, thread_count, shared_memory_size>>>(chunk_group);
+  const int num_blocks = chunk_group.chunk_count;
+  size_t w_buffer_size = static_cast<int>(num_blocks * sizeof(TOut1) + sizeof(int));
+  size_t d_buffer_size = static_cast<int>(num_blocks * sizeof(TOut2) + sizeof(int));
+
+  ORT_ENFORCE(w_buffer_size*sizeof(TOut1) + d_buffer_size*sizeof(TOut2) <= reduction_buffer_size);
```

Comment:
maybe i'm missing something - it seems like w/d_buffer_size are in bytes already, why do we multiply by sizeof(TOut1/2) here?
also, i guess w_buffer_size has space for an extra int for use as the block completion counter. why does d_buffer_size also include an extra int?
do we need to account for alignment of TOut1's, TOut2's and the int?